pkg/controller/utils/utils.go

```diff
@@ -107,12 +112,40 @@ func CheckCloudCredCreation(awsClient ccaws.Client, logger log.FieldLogger) (boo
 	return CheckPermissionsAgainstActions(awsClient, credMintingActions, logger)
 }
 
+// getClientDetails will return the *iam.User associated with the provided client's credentials,
+// a boolean indicating whether the user is the 'root' account, and any error encountered
+// while trying to gather the info.
+func getClientDetails(awsClient ccaws.Client) (*iam.User, bool, error) {
+	rootUser := false
+
+	user, err := awsClient.GetUser(nil)
+	if err != nil {
+		return nil, rootUser, fmt.Errorf("error querying username: %v", err)
+	}
+
+	// Detect whether the AWS account's root user is being used
+	parsed, err := arn.Parse(*user.User.Arn)
+	if err != nil {
+		return nil, rootUser, fmt.Errorf("error parsing user's ARN: %v", err)
+	}
+	if parsed.AccountID == *user.User.UserId {
+		rootUser = true
+	}
+
+	return user.User, rootUser, nil
+}
+
 // CheckPermissionsUsingQueryClient will use queryClient to query whether the credentials in targetClient can perform the actions
 // listed in the statementEntries. queryClient will need iam:GetUser and iam:SimulatePrincipalPolicy
 func CheckPermissionsUsingQueryClient(queryClient, targetClient ccaws.Client, statementEntries []minterv1.StatementEntry, logger log.FieldLogger) (bool, error) {
-	targetUsername, err := targetClient.GetUser(nil)
+	targetUser, isRoot, err := getClientDetails(targetClient)
 	if err != nil {
-		return false, fmt.Errorf("error querying current username: %v", err)
+		return false, fmt.Errorf("error gathering AWS credentials details: %v", err)
+	}
+	if isRoot {
+		// warn about using the root creds, and just return that the creds are good enough
+		logger.Warn(rootUserDetected)
```

Comment:
We should probably keep the log message with the log statement. I assume at one point you planned to re-use the message somewhere, maybe a condition? Did that get scrapped for some reason?

Reply:
yeah, there was a version where i was checking both the queryClient and the targetClient for 'root', but i realized it wasn't necessary for the queryClient. i'll update it